torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3601,7 +3601,10 @@ def no_sync(self) -> Generator:
             will only be offloaded right after the eventual sync.
         """
         self._lazy_init()
-        assert self._is_root, "`no_sync()` on inner FSDP instances is not supported"
+        if not self._is_root:
+            raise RuntimeError(
+                "`no_sync()` on inner FSDP instances is not supported. Please call `no_sync()` on root FSDP module."
```

Comment:
nit: In case you are okay with re-pushing:
```suggestion
                "`no_sync()` on inner FSDP instances is not supported. Please call `no_sync()` on the root FSDP module."
```